app/main.py

```diff
@@ -1,13 +1,21 @@
 def format_linter_error(error: dict) -> dict:
-    # white your code here
-    pass
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # white your code here
-    pass
+    return {
+        "errors": [format_linter_error(i) for i in errors],
+        "path": file_path,
+        "status": "passed" if errors == [] else "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # white your code here
-    pass
+    return [format_single_linter_file(key, value)
+            for key, value in linter_report.items()]
```

Comment:
what are key and value?

Reply:
key - it is the name of the dictionary list. like("./source_code_2.py": [])
value - it is the each key: value  in the dict. like ( "line": 112, "column": 6, "message": "no newline at end of file", "name": "W292", "source": "flake8",)

Reply:
your names are uninformative - better use `file_path` and `errors`

Reply:
yes with these names I can't explain the answer

Reply:
half an hour I thought how to explain these names

Reply:
make commit please

Reply:
you don't need to explain these names
you should refactor your code
```
    return [format_single_linter_file(key, value)
            for file_path, errors in linter_report.items()]
```

Reply:
i made new commit 
'replaced the names in (def format_linter_report()) key - faile_path, value - errors'

Reply:
Thank you, I will work on my mistakes